### PR TITLE
Test cases for #9525, #9526 and #9527

### PR DESCRIPTION
**Pull request by JanKoehnlein**

### CODE REVIEW

File: masl/test/9432_enummore/test2.svc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  primaries := PrimaryColour.red;              // Explicit Type scope. No ambiguity.
+  primaries := Colouring::red;                 // Explicit domain scope. Ambiguity resolved by examining the assigned to type.
+  primaries := red;                            // No scope. Ambiguity resolved by examining the assigned to type.
+
+  shapes := Shapes::Shape.square;              // Explicit domain and type scope. No ambiguity.
+  shapes := Shapes::square;                    // Explicit domain scope. Ambiguity resolved by examining the assigned to type.
+
+  quads  := Shapes::Quadrilateral.square;      // Explicit domain and type scope. No ambiguity.
+  quads  := Shapes::square;                    // Explicit domain scope. Ambiguity resolved by examining the assigned to type.
+end;
```
Comment (keithbrown) [inline masl/test/9432_enummore/test2.svc]: should be "end service;"

File: masl/test/9525_emptyStatement/EmptyStatement.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
+domain emptyStatement is 
+	service test();
+end;
```
Comment (keithbrown) [inline masl/test/9525_emptyStatement/EmptyStatement.mod]: should be "end domain;"

File: masl/test/9525_emptyStatement/EmptyStatement.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+domain emptyStatement is 
+	service test();
```
Comment (keithbrown) [inline masl/test/9525_emptyStatement/EmptyStatement.mod]: add "public" so this test case can be used in round trip testing

File: masl/test/9525_emptyStatement/test.svc [hunk 4]
```diff
@@ -0,0 +1,8 @@
+service emptyStatement::test() is
```
Comment (keithbrown) [inline masl/test/9525_emptyStatement/test.svc]: add public

File: masl/test/9525_emptyStatement/test.svc [hunk 5]
```diff
@@ -0,0 +1,8 @@
+service emptyStatement::test() is 
+	a : integer;
+begin
+  ;
+  a := 1;;;
+  ;;
+
+end;
```
Comment (keithbrown) [inline masl/test/9525_emptyStatement/test.svc]: "end service;"

File: masl/test/9526_dictionary/dictionary.mod [hunk 6]
```diff
@@ -0,0 +1,3 @@
+domain dict is 
+	service test();
```
Comment (keithbrown) [inline masl/test/9526_dictionary/dictionary.mod]: "public service" and "end domain"

File: masl/test/9526_dictionary/test.svc [hunk 7]
```diff
@@ -0,0 +1,15 @@
+service dict::test() is
```
Comment (keithbrown) [inline masl/test/9526_dictionary/test.svc]: "public"

File: masl/test/9526_dictionary/test.svc [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+begin
+  sparseArray[i] := s;
+  s := sparseArray[i];
+
+  b := sparseArray'contains(3); 
+  // b := sparseArray'contains("hello"); // ERROR 
+
+  i := sparseArray'length();
+  erase sparseArray[i];
+end;
```
Comment (keithbrown) [inline masl/test/9526_dictionary/test.svc]: "end service"

File: masl/test/9532_durationCharacteristics/duration.mod [hunk 9]
```diff
@@ -0,0 +1,3 @@
+domain duration is
+service test();
```
Comment (keithbrown) [inline masl/test/9532_durationCharacteristics/duration.mod]: "public service" and "end domain"

File: masl/test/9532_durationCharacteristics/test.svc [hunk 10]
```diff
@@ -0,0 +1,11 @@
+service duration::test() is
```
Comment (keithbrown) [inline masl/test/9532_durationCharacteristics/test.svc]: "public service"

File: masl/test/9532_durationCharacteristics/test.svc [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+  d : duration;
+begin
+  d := duration'days(1.5);
+  d := duration'hours(1.5);
+  d := duration'minutes(1.5);
+  d := duration'seconds(1.5);
+  d := duration'milliseconds(1.5);
+  d := duration'microseconds(1.5);
+  d := duration'nanoseconds(1.5);
+end;
```
Comment (keithbrown) [inline masl/test/9532_durationCharacteristics/test.svc]: "end service"

File: masl/test/9533_dictionaryIndex/dictionary.mod [hunk 12]
```diff
@@ -0,0 +1,3 @@
+domain dict2 is 
+	service test();
```
Comment (keithbrown) [inline masl/test/9533_dictionaryIndex/dictionary.mod]: "public service" and "end domain"

File: masl/test/9533_dictionaryIndex/test.svc [hunk 13]
```diff
@@ -0,0 +1,15 @@
+service dict2::test() is
```
Comment (keithbrown) [inline masl/test/9533_dictionaryIndex/test.svc]: "public service"

File: masl/test/9533_dictionaryIndex/test.svc [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+begin
+  lookup1[s] := s;
+  s := lookup1[s];
+
+  lookup2[s] := s;
+  s := lookup2[s];
+
+  lookup3[s] := s;
+  s := lookup3[s];
+end;
```
Comment (keithbrown) [inline masl/test/9533_dictionaryIndex/test.svc]: "end service"

File: masl/test/9582_performance/domain.mod [hunk 15]
```diff
@@ -0,0 +1,3 @@
+domain perf is
+	service foo() return integer;
```
Comment (keithbrown) [inline masl/test/9582_performance/domain.mod]: "public service" and "end domain"

File: masl/test/9582_performance/foo.svc [hunk 16]
```diff
@@ -0,0 +1,9 @@
+service perf::foo() return integer is
```
Comment (keithbrown) [inline masl/test/9582_performance/foo.svc]: "public service"

File: masl/test/9582_performance/foo.svc [hunk 17]
```diff
@@ -0,0 +1,9 @@
+service perf::foo() return integer is 
+  i : integer;
+begin
+  i := foo() + foo() + foo() + foo() + foo() + 
+       foo() + foo() + foo() + foo() + foo() + 
+       foo() + foo() + foo() + foo() + foo() +
+       foo() + foo() + foo() + foo() + foo();   
+  return 1;     
+end;
```
Comment (keithbrown) [inline masl/test/9582_performance/foo.svc]: "end service"

**Comment by JanKoehnlein** [comment]
Why are all these changes necessary? They examples work fine in the IDE as they are.

**Comment by keithbrown** [comment]
We have a tool "masl_round_trip" that takes an input masl model, converts it to xtuml, imports it into BridgePoint, then exports masl back out and compares it with the original input data.  It expects to find zero diffs, proving our flow worked properly.  I will be including these test models you've created in our round trip regression suite.

Messages not specified with a visibility are output by BridgePoint as "private", which causes a diff in the round trip report.  Specifying a visibility makes the diff report clean.  I do not know why the UI simply uses "end", I'll have to ask around to find that out.  But I do know that the MASL exported by BridgePoint says "end domain;" and "end service;".

**Pull request merged**